Assets/Scripts/Inge/TeleType.cs

```diff
@@ -0,0 +1,57 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using TMPro;
+using System;
+
+public class TeleType : MonoBehaviour
+{
+    private TextMeshProUGUI m_textMeshPro;
+    private bool isFirstTime;
+
+    public float speed;
+
+    void Start()
+    {
+        
+    }
+
+    private void OnEnable()
+    {
+        isFirstTime = true;
+        m_textMeshPro = gameObject.GetComponent<TextMeshProUGUI>() ?? gameObject.AddComponent<TextMeshProUGUI>();
```

Comment:
```suggestion
        m_textMeshPro = gameObject.GetComponent<TextMeshProUGUI>();
```
Boven aan de klasse de flag `[RequireComponent(typeof(TextMeshProUGUI))]` toevoegen om er zeker van te zijn dat dit script afhankelijk is van TextMeshProUGUI